docs/src/implementation.md

```diff
@@ -2,20 +2,20 @@
 
 `CategoricalArray` is made of the two fields:
 
-- `refs`: an integer array that stores the position of the category level in the `index` field of `CategoricalPool` for each `CategoricalArray` element; `0` denotes a missing value (for `CategoricalArray{Union{T, Null}}` only).
+- `refs`: an integer array that stores the position of the category level in the `index` field of `CategoricalPool` for each `CategoricalArray` element; `0` denotes a missing value (for `CategoricalArray{Union{T, Missing}}` only).
 - `pool`: the `CategoricalPool` object that maintains the levels of the array.
 
 The `CategoricalPool{V,R,C}` type keeps track of the levels of type `V` and associates them with an integer reference code of type `R` (for internal use). It offers methods to set the levels, change their order while preserving the references, and efficiently get the integer index corresponding to a level and vice-versa. Whether the values of `CategoricalArray` are ordered or not is defined by an `ordered` field of the pool. Finally, `CategoricalPool` keeps a `valindex` vector of value objects of type `C` (`CategoricalString{R}` or `CategoricalValue{V, R}`), so that `getindex` can return the existing object instead of allocating a new one.
 
 The type parameters of `CategoricalArray{T, N, R <: Integer, V, C, U}` are a bit complex:
- - `T` is the type of array elements without `CategoricalString`/`CategoricalValue` wrappers; if `T >: Null`, then the array is nullable.
+ - `T` is the type of array elements without `CategoricalString`/`CategoricalValue` wrappers; if `T >: Missing`, then the array supports missing values.
  - `N` is the number of array dimensions.
  - `R` is the reference type, the element type of the `refs` field; it allows optimizing memory usage depending on the number of levels (i.e. `CategoricalArray` with less than 256 levels can use `R = UInt8`).
- - `V` is the type of the levels, it is equal to `T` for non-nullable arrays; for nullable arrays, `T = Union{V, Null}`
- - `C` is the type of categorical values, i.e. the objects returned when indexing non-null elements of `CategoricalArray`. `C` is `CategoricalString{R}` if `V = String` and `CategoricalValue{V, R}` for any other `V`.
- - `U` can be either `Union{}` for non-nullable arrays, or `Null` for the nullable ones.
+ - `V` is the type of the levels, it is equal to `T` for arrays which do not support missing values; for arrays which support missing values, `T = Union{V, Missing}`
+ - `C` is the type of categorical values, i.e. the objects returned when indexing non-missing elements of `CategoricalArray`. `C` is `CategoricalString{R}` if `V = String` and `CategoricalValue{V, R}` for any other `V`.
+ - `U` can be either `Union{}` for arrays which do not support missing values, or `Missing` for those who support them.
```

Comment:
`who support them` reads a little funny. Maybe use `those which` as you used above and below, or be more explicit and say `arrays which`?

Reply:
Good catch!